src/main/java/com/yahoo/memory/StepBoolean.java

```diff
@@ -5,44 +5,28 @@
 
 package com.yahoo.memory;
 
-import java.util.concurrent.atomic.AtomicBoolean;
-
 /**
  * This is a step boolean function that can change its state only once and is thread-safe.
  *
  * @author Lee Rhodes
  */
 final class StepBoolean {
-  private final boolean initial;
-  private AtomicBoolean state = new AtomicBoolean(false);
+  private volatile boolean state;
 
   StepBoolean(final boolean initialState) {
-    this.initial = initialState;
-    this.state.set(initialState);
+    this.state = initialState;
   }
 
   /**
    * Gets the current state.
    * @return the current state.
    */
   boolean get() {
-    return this.state.get();
+    return this.state;
   }
 
-  /**
-   * This changes the state of this step boolean function if it has not yet changed.
-   * If the state has already changed this does nothing and returns false.
-   * @return true if the state changed due to this operation
-   */
-  boolean change() {
```

Comment:
By eliminating change() and hasChanged(), the user must keep track of what the initial state was.  The only important thing to know about a step-boolean is whether the state has in-fact changed.